lib/lightning/kafka_triggers/message_worker.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Lightning.KafkaTriggers.MessageWorker do
+  @moduledoc """
+  Requests a message from the MessageServer and manages the state of the
+  returned message.
+  """
+  use GenServer
+
+  alias Lightning.KafkaTriggers.MessageHandling
+  alias Lightning.KafkaTriggers.MessageServer
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  @impl true
+  def init(opts) do
+    Process.send_after(self(), :request_message, 1000)
+
+    {:ok, opts}
+  end
+
+  @impl true
+  def handle_info(:request_message, state) do
+    next_message_delay = Keyword.fetch!(state, :next_set_delay)
+    no_message_delay = Keyword.fetch!(state, :no_set_delay)
+
+    case MessageServer.next_message() do
+      nil ->
+        Process.send_after(self(), :request_message, no_message_delay)
+
+      message_id ->
+        MessageHandling.process_message_for(message_id)
```

Comment:
Same here, there is a real setup where `process_message_for` is called and can be tested.

Reply:
@jyeshe  This method shares some logic with `process_candidate_for` (`MessageHandling.create_work_order/1` and `MessageHandling.maybe_delete_candidate`), so it felt more coherent to keep this method there given that the primary goal is to first see if it works when drinking from the waterfall :).

I could certainly see an outcome where each server type passes in a finder query to a generalised `MessageHandling.process_message_for` method - that would be cleaner. 

However, I would like to ask for some leeway to do that - I believe that an important part of the load testing is to test the end-to-end solution, and we cannot see that until the Kafka message processing is stable.  So, I would like to confirm that this change (along with another PR yet to be crafted) delivers that and then I am happy to improve the engineering quality.

Reply:
Even though it's closely related to the `MessageWorker` I understand the reason you've put it on the `MessageHandling`. It's just part of the effort to make it simpler once the requirements already brings complexity to the code (from the need to have separate processing lanes for each key and now without keys).